lib/evmone/eof.cpp

```diff
@@ -167,12 +195,19 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_eof_headers(bytes_v
     if (state != State::terminated)
         return EOFValidationError::section_headers_not_terminated;
 
-    const auto section_bodies_size = section_headers[TYPE_SECTION].front() +
-                                     std::accumulate(section_headers[CODE_SECTION].begin(),
-                                         section_headers[CODE_SECTION].end(), 0) +
-                                     section_headers[DATA_SECTION].front();
+    const auto section_bodies_without_data =
+        section_headers[TYPE_SECTION].front() +
+        std::accumulate(
+            section_headers[CODE_SECTION].begin(), section_headers[CODE_SECTION].end(), 0) +
+        std::accumulate(section_headers[CONTAINER_SECTION].begin(),
+            section_headers[CONTAINER_SECTION].end(), 0);
     const auto remaining_container_size = container_end - it;
-    if (section_bodies_size != remaining_container_size)
+    // Only data section may be truncated, so remaining_container size must be in
```

Comment:
We need verify we are validating subcontainers ([here](https://github.com/ethereum/evmone/blob/3ebb3c3dc8df344c62853950be8688dd99e333eb/lib/evmone/eof.cpp#L595)), because for regular EOF code validation ([here](https://github.com/ethereum/evmone/blob/3ebb3c3dc8df344c62853950be8688dd99e333eb/lib/evmone/eof.cpp#L761)) data section must be the exact size indicated in the data section header

Reply:
Makes sense, eofparse tool would also need to validate the container is not truncated as well.

Reply:
> Makes sense, eofparse tool would also need to validate the container is not truncated as well.

I would say it should check that all CREATE3 use non-truncated container, because it is part of validation, but top-level container should be allowed to be truncated.